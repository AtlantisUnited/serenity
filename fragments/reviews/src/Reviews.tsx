import React                        from 'react'

import { Avatar }                   from '@ui/avatar'
import { Box, Column, Layout, Row } from '@ui/layout'
import { Rating }                   from '@ui/rating'
import { Text }                     from '@ui/text'

export const Reviews = ({ reviews = [] }) => (
  <Column>
    {reviews.map((review: any) => (
      <Box
        key={review.id}
        my='7px'
        p='16px 25px 25px 16px'
        pr={['16px', '25px']}
        width='100%'
        border='black'
        borderRadius='extra'
      >
        <Layout>
          <Avatar
            width={56}
            height={56}
            color='alto'
            src={(review.customer.photo && review.customer.photo.url) || null}
          />
        </Layout>
        <Layout flexBasis={16} flexShrink={0} />
        <Layout flexGrow={1}>
          <Column>
            <Layout flexBasis='7px' />
            <Layout>
              <Row justifyContent='space-between' alignItems='center'>
                <Layout>
                  <Text fontWeight='medium' fontSize='regular' lineHeight='medium'>
                    {review.customer.personalInformation.firstName}{' '}
                    {review.customer.personalInformation.lastName}
                  </Text>
                </Layout>
                <Layout flexBasis={20} />
                <Layout>
                  <Rating rating={review.rating} />
                </Layout>
              </Row>
            </Layout>
            <Layout flexBasis={32} />
            <Layout>
              <Text lineHeight='medium'>{review.comment}</Text>
            </Layout>
          </Column>
        </Layout>
      </Box>
    ))}
  </Column>
)
