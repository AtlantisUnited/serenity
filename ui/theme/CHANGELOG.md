# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.5 (2021-05-10)

**Note:** Version bump only for package @ui/theme

## 0.0.4 (2020-09-28)

### Bug Fixes

- **accounts:** add emotion-server, delete font ([08acf5a](https://github.com/Atlantis-Lab/serenity/commit/08acf5a7169352a606925fdfbbaa18f9917a8106))

## 0.0.3 (2020-06-30)

**Note:** Version bump only for package @ui/theme

## 0.0.2 (2020-06-10)

**Note:** Version bump only for package @ui/theme
