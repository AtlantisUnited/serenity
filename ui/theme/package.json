{
  "name": "@ui/theme",
  "version": "0.0.5",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@types/react": "^17.0.6",
    "@types/styled-system": "5.1.4"
  },
  "dependencies": {
    "@emotion/css": "^11.1.3",
    "@emotion/react": "^11.4.0",
    "@emotion/styled": "^11.3.0",
    "react": "^17.0.2",
    "styled-system": "^5.1.0"
  }
}
