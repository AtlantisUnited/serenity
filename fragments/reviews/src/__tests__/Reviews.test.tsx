import React             from 'react'

import { ThemeProvider } from '@ui/theme'
import { render }        from '@testing-library/react'

import { Reviews }       from '../Reviews'

describe('Reviews', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <ThemeProvider>
          <Reviews />
        </ThemeProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
