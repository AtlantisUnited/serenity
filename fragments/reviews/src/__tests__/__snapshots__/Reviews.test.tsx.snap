// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    class="emotion-0"
    display="flex"
    width="100%"
  />
</DocumentFragment>
`;
